### DISCUSSION TITLE
Opening Preview in Chrome or Firefox from application in WSL2?

### DISCUSSION THREAD

**Discussion by ajmas** [forwardemail/forwardemail.net#256] [Q&A] [answered]
Does anyone know how to have an application running in WSL2 have the email preview open up in Chrome or Firefox of the host system?

It is for an application using this package: https://www.npmjs.com/package/email-templates

Note, this works fine on macOS, but not when running in WSL2.

**Comment by titanism** [comment, marked as answer]
Hi there, WSL2 most likely is running a VM, and that doesn't have access to host machine.  Also, this isn't the place to ask your question.  Your question should belong in the <https://github.com/forwardemail/test-preview-emails-cross-browsers-ios-simulator-nodejs-javascript> repository since that is with regards to `preview-email` package.

**Reply by titanism** [reply]
@ajmas See <https://github.com/sindresorhus/open/issues/174#issuecomment-1689551087>.  You'd need to have your `email-templates` instance, e.g. `const email = new EmailTemplates({ ... })` have an option in the `{ ... }` part of:

```
{
  previewEmail: {
    open: {
      urlTransform: (path: string) => `file://\\\\wsl.localhost/${process.env.WSL_DISTRO_NAME}${path}`
    }
  }
}
```
